testing/integration/otel_test.go

```diff
@@ -1099,3 +1100,191 @@ service:
 	fixtureWg.Wait()
 	require.True(t, err == nil || errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded), "Retrieved unexpected error: %s", err.Error())
 }
+
+func TestFBOtelRestartE2E(t *testing.T) {
+	// This test ensures that filebeatreceiver is able to deliver logs even
+	// in advent of a collector restart.
+	// The input is a file that is being appended to n times during the test.
+	// It starts a filebeat receiver and then restarts it a couple times.
+	// At the end it asserts that the number of logs in ES is equal to the number of
+	// lines in the input file.
+	info := define.Require(t, define.Requirements{
+		Group: Default,
+		Local: true,
+		OS: []define.OS{
+			{Type: define.Windows},
+			{Type: define.Linux},
+			{Type: define.Darwin},
+		},
+		Stack: &define.Stack{},
+	})
+	tmpDir := t.TempDir()
+
+	inputFile, err := os.CreateTemp(tmpDir, "input.txt")
+	require.NoError(t, err, "failed to create temp file to hold data to ingest")
+	inputFilePath := inputFile.Name()
+
+	// Create the otel configuration file
+	type otelConfigOptions struct {
+		InputPath  string
+		HomeDir    string
+		ESEndpoint string
+		ESApiKey   string
+		Index      string
+	}
+	esEndpoint, err := getESHost()
+	require.NoError(t, err, "error getting elasticsearch endpoint")
+	esApiKey, err := createESApiKey(info.ESClient)
+	require.NoError(t, err, "error creating API key")
+	require.True(t, len(esApiKey.Encoded) > 1, "api key is invalid %q", esApiKey)
+	index := "logs-integration-default"
+	otelConfigTemplate := `receivers:
+  filebeatreceiver:
+    filebeat:
+      inputs:
+        - type: filestream
```

Comment:
This is perhaps a bit beyond the scope of this test, but it would be interesting to confirm that the registry is actually updating as expected and we aren't re-reading the file from the beginning every time we restart.

This would validate that acknowledgements flow back to the receiver properly.

This would perhaps best be done as a separate test that writes a file, confirms we read it all, writes a single additional line, then confirms only that one line is resent.